plugins/inputs/statsd/statsd_test.go

```diff
@@ -1810,6 +1817,53 @@ func TestParse_InvalidAndRecoverIntegration(t *testing.T) {
 			"cpu_time_idle",
 			map[string]string{
 				"metric_type": "counter",
+				"temporality": "cumulative",
+			},
+			map[string]interface{}{
+				"value": 42,
+			},
+			time.Now(),
+			telegraf.Counter,
+		),
+	}
+	testutil.RequireMetricsEqual(t, expected, acc.GetTelegrafMetrics(), testutil.IgnoreTime())
+
+	require.NoError(t, conn.Close())
+}
+
+func TestParse_DeltaCounter(t *testing.T) {
+	statsd := Statsd{
+		Log:                    testutil.Logger{},
+		Protocol:               "tcp",
+		ServiceAddress:         "localhost:8125",
+		AllowedPendingMessages: 10000,
+		MaxTCPConnections:      250,
+		TCPKeepAlive:           true,
+		// Delete Counters causes Delta temporality to be added
+		DeleteCounters: true,
+	}
+
+	acc := &testutil.Accumulator{}
+	require.NoError(t, statsd.Start(acc))
+	defer statsd.Stop()
+
+	addr := statsd.TCPlistener.Addr().String()
+	conn, err := net.Dial("tcp", addr)
+	require.NoError(t, err)
+
+	_, err = conn.Write([]byte("cpu.time_idle:42|c\n"))
+	require.NoError(t, err)
+
+	time.Sleep(100 * time.Millisecond)
+	require.NoError(t, statsd.Gather(acc))
+	acc.Wait(1)
```

Comment:
How about the following
```suggestion
        require.Eventuallyf(t, func() bool {
		require.NoError(t, statsd.Gather(acc))
                acc.Lock()
                defer acc.Unlock()
                return acc.NMetrics() >= 1
        }, time.Second, 100*time.Millisecond, "Expected 1 metric found %d", acc.NMetrics())
```
This prevents an infinite blocking of the test (which `acc.Wait` will do otherwise)...